community_usecase/Profile_generation/run_profile_generation.py

```diff
@@ -0,0 +1,394 @@
+    """Extract and logger.info every URL that the agents actually browsed."""
+
+    final_summary = None
+
+    for item in reversed(chat_history):
+        assistant_content = item.get("assistant", "")
+        if (
+                isinstance(assistant_content, str)
+                and "FINAL SUMMARY REPORT" in assistant_content
+        ):
+            final_summary = assistant_content
+    if final_summary:
+        logger.info("\033[94m===== RAW SUMMARY FROM AGENTS =====\033[0m")
+        logger.info(final_summary)
+    else:
+        logger.info(
+            "\033[91mNo final summary report found in chat history.\033[0m")
+
+    queried_urls: list[str] = []
+    for turn in chat_history:
+        for call in turn.get("tool_calls", []):
+            if call.get("tool_name") == "browse_url":
+                url = call.get("args", {}).get("start_url")
+                if url:
+                    queried_urls.append(url)
+
+    logger.info("\n============ URL Analysis ============")
+
+    logger.info(f"Total queried URLs: {len(queried_urls)}")
+    for url in queried_urls:
+        logger.info(url)
+    with open("process_history.json", "w", encoding="utf-8") as fp:
+        json.dump(chat_history, fp, ensure_ascii=False, indent=2)
+    logger.info("Records saved to process_history.json")
+    logger.info("============ Analysis Complete ============\n")
+
+    return final_summary
+
+
+# ------------------ Pydantic Schemas ------------------
+
+class PersonalInformation(BaseModel):
+    name: str = Field(description="Full name of the scholar")
+    positions: list[str] = Field(
+        description="List of current academic or professional positions")
+    contact_information: Optional[str] = Field(
+        description="Primary contact information such as email address or "
+                    "phone number"
+    )
+    social_media: Optional[list[str]] = Field(
+        description="List of social media or personal profile URLs (e.g., "
+                    "LinkedIn, Twitter)"
+    )
+    research_keywords: list[str] = Field(
+        description="Keywords summarizing key research areas or topics")
+    short_introduction: str = Field(
+        description="A brief one-paragraph summary of the scholar's profile")
+
+
+class Biography(BaseModel):
+    biography: str = Field(
+        description="Narrative biography including career trajectory and "
+                    "research journey")
+
+
+class ResearchInterests(BaseModel):
+    areas: list[str] = Field(
+        description="List of active or long-term research topics")
+
+
+class AwardsAndDistinctions(BaseModel):
+    honors: list[str] = Field(
+        description="List of awards, honors, and professional distinctions "
+                    "received")
+
+
+class Education(BaseModel):
+    degrees: list[str] = Field(
+        description="Academic degrees in reverse chronological order, "
+                    "including institution and year")
```

Comment:
The format problem can be solve by adding examples in the `Field` `description`.